.github/workflows/build_and_test_cmake.yml

```diff
@@ -0,0 +1,441 @@
+
+  linux:
+    runs-on: ubuntu-20.04
+    container: seladb/${{ matrix.image }}
+    strategy:
+      matrix:
+        include:
+          - image: ubuntu2204
+            python: python3
+            config-zstd: OFF
+          - image: ubuntu2004
+            python: python3
+            config-zstd: OFF
+          - image: ubuntu2004-zstd
+            python: python3
+            config-zstd: ON
+          - image: ubuntu1804
+            python: python3.8
+            config-zstd: OFF
+          - image: centos7
+            python: python3.8
+            test-flags: --test-args="-s"
+            config-zstd: OFF
+          - image: fedora34
+            python: python3
+            config-zstd: OFF
+          - image: alpine315
+            python: python3
+            config-zstd: OFF
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: "0"
+
+      - name: Mark repository as safe (Checkout actions is not run with same UID)
+        run: |
+          git config --global --add safe.directory /__w/PcapPlusPlus/PcapPlusPlus
+          git config --global --add safe.directory ${{ github.workspace }}
+
+      - name: Configure PcapPlusPlus
+        run: cmake -DLIGHT_PCAPNG_ZSTD=${{ matrix.config-zstd }} -S . -B $BUILD_DIR
+
+      - name: Build PcapPlusPlus
+        run: cmake --build $BUILD_DIR -j
+
+      - name: Test PcapPlusPlus
+        run: |
+          ${{ matrix.python }} -m pip install -U pip
+          ${{ matrix.python }} -m pip install -r ci/run_tests/requirements.txt
+          ${{ matrix.python }} ci/run_tests/run_tests.py --interface eth0 ${{ matrix.test-flags }}
+
+      - name: Test Examples
+        run: |
+          cd Tests/ExamplesTest
+          ${{ matrix.python }} -m pip install -U pip
+          ${{ matrix.python }} -m pip install -r requirements.txt
+          ${{ matrix.python }} -m pytest --interface eth0 --root-path=../../Dist/examples_bin
+
+      - name: Check installation
+        run: cmake --install $BUILD_DIR
+
+      - name: Build Tutorials
+        run: |
+          mkdir -p build_examples
+          cmake -DPCAPPP_BUILD_TUTORIALS=ON -S Examples -B build_examples
+          cmake --build build_examples -j
+
+      - name: Test Tutorials
+        run: |
+          # We need to execute the HelloWorld in the same folder as the PCAP file
+          cp build_examples/Tutorials/Tutorial-HelloWorld/Tutorial-HelloWorld Examples/Tutorials/Tutorial-HelloWorld/
+          cd Examples/Tutorials/Tutorial-HelloWorld && ./Tutorial-HelloWorld
+
+  dpdk:
+    runs-on: ubuntu-20.04
+    container: seladb/${{ matrix.image }}
+    strategy:
+      matrix:
+        image:
+          [
+            ubuntu2004-dpdk2111,
+            ubuntu2004-dpdk2011,
+            ubuntu2004-dpdk1911,
+            ubuntu2004-dpdk1811,
+          ]
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: "0"
```

Comment:
why is this needed (here and in the other jobs)? 🤔

Reply:
this is needed also in other jobs to get the git history and grab the latest tag

Reply:
It's probably related to this comment: https://github.com/seladb/PcapPlusPlus/pull/944#discussion_r960266842

Reply:
yes if we decide to take the version from the git tag we need to clone with the history to get all the tags.

I mark this issue as resolved and we can continue to discuss on the other comment